gtk/Application.cc

```diff
@@ -524,23 +527,17 @@ tr_rpc_callback_status Application::Impl::on_rpc_changed(
 namespace
 {
 
-sig_atomic_t global_sigcount = 0;
-gpointer sighandler_cbdata = nullptr;
+#ifdef G_OS_UNIX
 
-void signal_handler(int sig)
+gboolean signal_handler(gpointer user_data)
 {
-    if (++global_sigcount > 1)
-    {
-        signal(sig, SIG_DFL);
-        raise(sig);
-    }
-    else if (sig == SIGINT || sig == SIGTERM)
-    {
-        g_message(_("Got signal %d; trying to shut down cleanly. Do it again if it gets stuck."), sig);
-        gtr_actions_handler("quit", sighandler_cbdata);
-    }
+    g_message(_("Got termination signal, trying to shut down cleanly. Do it again if it gets stuck."));
```

Comment:
Yes.

Reply:
I may have misread the question. The proper answer is: it doesn't matter now since `g_warning()` is no longer called inside the actual signal handler.

Reply:
[I got the impression from your other response](https://github.com/transmission/transmission/pull/3967#issuecomment-1283480936 "Constructive feedback from 2022-10-19") that this short feedback should be reconsidered another bit.
Are there different development views involved according to the term “signal handling” by the programming interfaces “POSIX” and “GLib”? :thinking:

Reply:
POSIX signals and Glib signals are two completely different things which aren't related to each other. I'm sorry but I'd rather not dive into details here; I believe there's enough information on both topics out there if you're interested to know more.